core/store/data_input.cpp

```diff
@@ -172,25 +157,19 @@ size_t buffered_index_input::refill() {
     throw eof_error(); // read past eof
   }
 
-  if (!buf_) {
-    buf_ = memory::make_unique<byte_type[]>(buf_size_);
-    begin_ = end_ = buf_.get();
-    seek_internal(start_);
-  }
-
-  begin_ = buf_.get();
-  end_ = begin_ + read_internal(buf_.get(), data_size);
+  begin_ = buf_;
```

Comment:
```suggestion
  assert(buf_);
  begin_ = buf_;
```
As now we have no auto-allocation in case of no buffer so at least we need to assert we have it.